fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }))
+      .catch(err => console.error('err: ', err));
+  }
+
+  addTodo(newTodo) {
+    const todoInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    };
+    fetch('/api/todos', todoInit)
+      .then(res => {
+        if (res.status === 201) {
+          res.json()
+            .then(data => {
+              this.setState(state => ({
+                todos: state.todos.concat([data])
+              }));
+            });
+        }
+      });
+  }
+
+  toggleCompleted(todoId) {
+    const currentStatus = this.state.todos[todoId - 1].isCompleted;
```

Comment:
Can't use the id as an index, if something got deleted the ids will get further and further off from the indexes.

You can find an item in an array using `.find` or you can find an index of an item in an array using `.findIndex`